src/abstractdataframe/abstractdataframe.jl

```diff
@@ -218,7 +202,7 @@ eltypes(df)
 ```
 
 """
-eltypes(df::AbstractDataFrame) = map!(eltype, Vector{Type}(undef, size(df,2)), columns(df))
+eltypes(df::AbstractDataFrame) = eltype.(columns(df))
```

Comment:
I think that having `eltypes` does not hurt and it is a useful function.
`eltype.(columns(df))` is not very verbose but I think it would not obvious for newcomers.

What are the arguments for deprecating it?

Reply:
Nothing special, I just find `eltype.(columns(df))` more explicit and standard. But we can discuss this elsewhere.